swarm/storage/resource.go

```diff
@@ -825,3 +848,31 @@ func (self *ResourceHandler) keyDataHash(key Key, data []byte) common.Hash {
 	hasher.Write(data)
 	return common.BytesToHash(hasher.Sum(nil))
 }
+
+// if first byte is the start of a multihash this function will try to parse it
+// if successful it returns the length of multihash data, 0 otherwise
+func isMultihash(data []byte) int {
```

Comment:
Any reason not to use [`multihash.Decode`](https://github.com/multiformats/go-multihash/blob/master/multihash.go#L223-L259)?

Reply:
Yes, the package is not actually in use now in the repo. Until it is necessary to add it, let's do away with the extra bloat. This check is perfectly valid according to the spec, afaics.